backend/danswer/connectors/slack/utils.py

```diff
@@ -13,3 +20,54 @@ def get_message_link(
     return (
         f"https://{workspace}.slack.com/archives/{channel_id}/p{message_ts_without_dot}"
     )
+
+
+def make_slack_api_call_paginated(
+    call: Callable[..., SlackResponse],
+) -> Callable[..., list[dict[str, Any]]]:
+    """Wraps calls to slack API so that they automatically handle pagination"""
+
+    def paginated_call(**kwargs: Any) -> list[dict[str, Any]]:
+        results: list[dict[str, Any]] = []
+        cursor: str | None = None
+        has_more = True
+        while has_more:
+            for result in call(cursor=cursor, limit=_SLACK_LIMIT, **kwargs):
+                has_more = result.get("has_more", False)
+                cursor = result.get("response_metadata", {}).get("next_cursor", "")
+                results.append(cast(dict[str, Any], result))
+        return results
+
+    return paginated_call
+
+
+def make_slack_api_rate_limited(
+    call: Callable[..., SlackResponse], max_retries: int = 3
```

Comment:
Should the 3 be configurable, declared elsewhere? doesn't really matter tho

Reply:
I personally think having the default in the function is easier to follow. IMO don't need to have this be controllable by an env variable